File: backend-java/src/main/java/com/arvirtualtryon/services/ProductService.java
package com.arvirtualtryon.services;

import com.arvirtualtryon.models.Category;
import com.arvirtualtryon.dtos.ProductRequestDTO;
import com.arvirtualtryon.dtos.ProductResponseDTO;

import java.util.List;

public interface ProductService {
    List<ProductResponseDTO> getAllProducts();
    ProductResponseDTO getProductById(Long id);
    List<ProductResponseDTO> getProductsByCategory(Category category);
    ProductResponseDTO createProduct(ProductRequestDTO productRequestDTO);
    ProductResponseDTO updateProduct(Long id, ProductRequestDTO productRequestDTO);
    void deleteProduct(Long id);
    void addOrUpdateProductsFromDirectory(String directoryPath);
    Category findOrCreateCategory(String name);
}

File: frontend_web/src/services/productService.js
import { API_ENDPOINTS, apiRequest } from '../config/api';

export const productService = {

  getAllProducts: async () => {
    return await apiRequest(API_ENDPOINTS.PRODUCTS);
  },

  getProductsByCategory: async (category) => {
    return await apiRequest(`${API_ENDPOINTS.PRODUCTS}?category=${category}`);
  },

  getProductById: async (id) => {
    return await apiRequest(`${API_ENDPOINTS.PRODUCTS}/${id}`);
  },

  createProduct: async (productData) => {
    return await apiRequest(API_ENDPOINTS.PRODUCTS, {
      method: 'POST',
      body: JSON.stringify(productData),
    });
  },

  updateProduct: async (id, productData) => {
    return await apiRequest(`${API_ENDPOINTS.PRODUCTS}/${id}`, {
      method: 'PUT',
      body: JSON.stringify(productData),
    });
  },

  deleteProduct: async (id) => {
    return await apiRequest(`${API_ENDPOINTS.PRODUCTS}/${id}`, {
      method: 'DELETE',
    });
  },
};